pkg/apis/monitoring/v1/types.go

```diff
@@ -583,6 +581,41 @@ type PodMonitorSpec struct {
 	SampleLimit uint64 `json:"sampleLimit,omitempty"`
 }
 
+// PodMetricsEndpoint defines a scrapeable endpoint of a Kubernetes Pod serving Prometheus metrics.
+// +k8s:openapi-gen=true
+type PodMetricsEndpoint struct {
```

Comment:
The `PodMetricsEndpoint` for a `PodMonitor` copies the `Endpoint` struct used in a `ServiceMonitor`. This is used to dis-entangle both. Ok with this @brancz?